portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -66,16 +77,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Entity messageEntity = new Entity("Message");
     messageEntity.setProperty(TIME, timestamp);
     messageEntity.setProperty(NEWEST_MESSAGE, fullMessage);
+
     datastore.put(messageEntity);
 
     response.sendRedirect("/write-message.html");
   }
 
   private String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
-    if (value == null) {
+    if (value == " ") {
       return defaultValue;
     }
     return value;
   }
+
+  private int convertToInt(String beingconverted) {
+    int convertee;
+    try {
+      convertee = Integer.parseInt(beingconverted);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + beingconverted);
+      return -1;
```

Comment:
I suspect that your app will crash if you return -1. You could return a default value instead, such as 5?